### ISSUE TITLE
Myloader reports an error when restoring data, and ctrl + c cannot interrupt the linux session, the error message is: because real_db_name isn't found. [BUG]

### ISSUE THREAD

**Issue by zhangchuangcheng0807** [mydumper/mydumper#825]
**Describe the bug**
myloader restores the data backed up by mydumper and reports an error. It keeps outputting "** (myloader:24892): WARNING **:" information, and cannot interrupt the linux session through ctrl+c or ctrl+d. Detailed error message:
** (myloader:24892): WARNING **: 18:42:06.249: It was not possible to process file: xsy_relation.p_custom_data_2288_ext-schema.sql (1) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288-schema.sql (1) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

**To Reproduce**
Command executed:
* mydumper with all the parameters
mydumper -h 10.43.x.x -P 3310 -u mydumper -p mydumper --trx-consistency-only -B xsy_relation -T 'xsy_relation.p_custom_data_2288,xsy_relation.p_custom_data_2288_ext,xsy_relation.p_custom_data_2288_view' -v 3 -o /data1/db_backup/data_migration/test2/
* myloader with all the parameters
myloader -h 10.43.x.x -u mydumper -p mydumper -P 3310 -B isales_db_isales_001 -s xsy_relation --skip-definer -t 4 -o -e -v 3 -d /data1/db_backup/data_migration/test2/

What mydumper and myloader version has been used?
mydumper 0.12.7-1, built against MySQL 5.7.39-42
myloader 0.12.7-1, built against MySQL 5.7.39-42

**Expected behavior**

**Log**
If applicable, add the --verbose 3 / -v 3 and --logfile / -L <filename> to the execution of the command and upload the file.
** (myloader:24892): WARNING **: 18:42:06.249: It was not possible to process file: xsy_relation.p_custom_data_2288_ext-schema.sql (1) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288-schema.sql (1) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288-metadata (2) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288_view-schema.sql (1) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288_ext-metadata (2) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288_view-schema-view.sql (3) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288_ext-schema.sql (1) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288-schema.sql (1) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288-metadata (2) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288_view-schema.sql (1) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288_ext-metadata (2) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288_view-schema-view.sql (3) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288_ext-schema.sql (1) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288-schema.sql (1) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288-metadata (2) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

**Backup**
If applicable, add the list of content of the database path, an `ls -l` will be enough
[root@localhost test2]# mydumper -h 10.43.x.x -P 3310 -u mydumper -p mydumper --trx-consistency-only -B xsy_relation -T 'xsy_relation.p_custom_data_2288,xsy_relation.p_custom_data_2288_ext,xsy_relation.p_custom_data_2288_view' -v 3 -o /data1/db_backup/data_migration/test2/

[root@localhost test2]# pwd
/data1/db_backup/data_migration/test2
[root@localhost test2]# ll
total 76
-rw-r--r-- 1 root root   184 Sep 16 18:26 metadata
-rw-r--r-- 1 root root     1 Sep 16 18:26 xsy_relation.p_custom_data_2288_ext-metadata
-rw-r--r-- 1 root root  3947 Sep 16 18:26 xsy_relation.p_custom_data_2288_ext-schema.sql
-rw-r--r-- 1 root root     1 Sep 16 18:26 xsy_relation.p_custom_data_2288-metadata
-rw-r--r-- 1 root root 20098 Sep 16 18:26 xsy_relation.p_custom_data_2288-schema.sql
-rw-r--r-- 1 root root 11552 Sep 16 18:26 xsy_relation.p_custom_data_2288_view-schema.sql
-rw-r--r-- 1 root root 22123 Sep 16 18:26 xsy_relation.p_custom_data_2288_view-schema-view.sql
-rw-r--r-- 1 root root    99 Sep 16 18:26 xsy_relation-schema-create.sql

[root@localhost test2]# myloader -h 10.43.x.x -u mydumper -p mydumper -P 3310 -B isales_db_isales_001 -s xsy_relation --skip-definer -t 4 -o -e -v 3 -d /data1/db_backup/data_migration/test2/

 ** (myloader:24892): WARNING **: 18:42:06.249: It was not possible to process file: xsy_relation.p_custom_data_2288_ext-schema.sql (1) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288-schema.sql (1) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288-metadata (2) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288_view-schema.sql (1) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288_ext-metadata (2) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:24892): WARNING **: 18:42:06.250: It was not possible to process file: xsy_relation.p_custom_data_2288_view-schema-view.sql (3) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

**How to repeat**
If applicable, add the minimal table structure and data that we need to reproduce the issue.
Or upload a [core dump](https://github.com/mydumper/mydumper/wiki/Support#core-dump) for cases where mydumper or myloader is crashing.

**Environment (please complete the following information):**
 - OS version: [e.g. CentOS 7.9, Ubuntu 20.04]
 CentOS 7.5
 - MyDumper version: [ mydumper --version ]
[root@localhost ~]# mydumper --version
mydumper 0.12.7-1, built against MySQL 5.7.39-42
[root@localhost ~]# myloader --version
myloader 0.12.7-1, built against MySQL 5.7.39-42

**Additional context**

**Comment by aqueos** [comment]
hi,

 very same error here, upgrading to 0.12.7-1, and restore do not work anymore. Very same restore line was working just before the myloader upgrade.

t was not possible to process file: xxxxx.xxxxx.00000.sql (3) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

also had to ctrl-Z and kill -9 the process.

regards,
Ghislain.

version mydumper_0.12.7-1.buster_amd64.deb  on debian buster

**Comment by all-probros** [comment]
this also happened to me on macbook m1
myloader 0.12.7-1, built against MySQL 8.0.30

**Comment by davidducos** [comment]
Hi @zhangchuangcheng0807 / @aqueos / @all-probros , 
Just to clarify, WARNING are just warnings not ERRORs. 
About the ctrl+c issue, I need you to execute `show processlist` and confirm that there is no myloader execution, as I already commented on other issues, ctrl+c will not stop job executions, it will not continue executing new jobs. If you need to stop the execution, you can press `ctrl+\` 

About the WARNING message:
```It was not possible to process file: xsy_relation.p_custom_data_2288_ext-metadata (2) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported```
Since we merge stream and directory, we are not ordering the files, that is why we might need to renqueue files again. For instance, if this is a directory backup, OS gives us the list of the files and we read the metadata files first, our internal table structure has not been created yet, we renqueue that file to later processing.

**Comment by aqueos** [comment]
hi,

well on my side the warning keep running forever. I have a process that is running like this for minutes now. Still nothing done to the database.

 i have Thread=4 and 4 mysql connexion sleeping
| 20 | root | localhost | AQTEST | Sleep   |  201 |       | NULL                  |  201205 |         0 |             0 |
| 21 | root | localhost | AQTEST | Sleep   |  201 |       | NULL                  |  201204 |         0 |             0 |
| 22 | root | localhost | AQTEST | Sleep   |  201 |       | NULL                  |  201203 |         0 |             0 |
| 23 | root | localhost | AQTEST | Sleep   |  201 |       | NULL                  |  201202 |         0 |             0 |

Mon 19 Sep 2022 04:44:58 PM CEST
Mon 19 Sep 2022 04:49:01 PM CEST

i stopped it after 4 minutes of warning in loop.

myloader --database=AQTEST --directory=/var/backups/mysql/2022-09-18-23h01  --source-db=xxxxx --threads=1 --verbose=3 --overwrite-tables --user=root --serialized-table-creation --password=yyyy; 

** (myloader:26936): WARNING **: 16:49:01.187: It was not possible to process file: xxxx.ldpratique_typetarif-metadata (2) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:26936): WARNING **: 16:49:01.190: It was not possible to process file: xxxx.ldpratique_galerie-metadata (2) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:26936): WARNING **: 16:49:01.193: It was not possible to process file: xxx.structures_temporaires-metadata (2) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:26936): WARNING **: 16:49:01.196: It was not possible to process file: xxxx.geo_api_communes-metadata (2) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

** (myloader:26936): WARNING **: 16:49:01.199: It was not possible to process file: xxx.licence_surclassement_categorie_ages-metadata (2) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported

normaly it start restoring after a few seconds.

regards,
Ghislain.

**Comment by aqueos** [comment]
tried without --serialized-table-creationand thread=1 or 4 with the same effect.

**Comment by davidducos** [comment]
@aqueos do you have the schema-create file for those files? This file: xxxx-schema-create.sql ?

**Comment by aqueos** [comment]
yes, i sent you one by email. I prefer not having it public here :)

**Comment by zhangchuangcheng0807** [comment]
> Hi @zhangchuangcheng0807 / @aqueos / @all-probros , Just to clarify, WARNING are just warnings not ERRORs. About the ctrl+c issue, I need you to execute `show processlist` and confirm that there is no myloader execution, as I already commented on other issues, ctrl+c will not stop job executions, it will not continue executing new jobs. If you need to stop the execution, you can press `ctrl+\`
> 
> About the WARNING message: `It was not possible to process file: xsy_relation.p_custom_data_2288_ext-metadata (2) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported` Since we merge stream and directory, we are not ordering the files, that is why we might need to renqueue files again. For instance, if this is a directory backup, OS gives us the list of the files and we read the metadata files first, our internal table structure has not been created yet, we renqueue that file to later processing.

HI @davidducos, 
Thanks for replying to this question.
Three points to answer your questions above:
1. When using myloader to restore data, execute show processlist on the database instance; the statement does not find that myloader restores the data session (that is, there is no connection session of the mydumper user), and no data restore operation is performed.
2. I would like to explain again about data backup and recovery. The above operation is to back up two tables in the xsy_relation database: p_custom_data_2288, p_custom_data_2288_ext and a view: p_custom_data_2288_view; then restore the backup data to the isales_db_isales_001 database, that is, the source database and the The target library names are not the same. In the previous version v0.12.5-1, table data and view recovery can be performed normally, but in v0.12.7-1 version, it is not possible, and the above problem occurs.
3. I just tested that the warning log input can be aborted and the myloader recovery command can be aborted through the crtl+\ command.

**Comment by davidducos** [comment]
@zhangchuangcheng0807 , I found what is causing your issue. You are using -s and -B which I was not taking into account. We must add some test for this scenarios. I might be able to release a hot fix later today.

@aqueos I need you to send me your mydumper and myloader commands and an `ls -l` of the backup directory.

**Comment by davidducos** [comment]
@aqueos based on the info that you send me, your problem is the same than @zhangchuangcheng0807.
Both can use the PR that I already merge, I might be able to release the patch tomorrow.

**Comment by aqueos** [comment]
thanks again for your quick action, if you rebuild the package i will be able to test it for you in the same condition.
thanks for the hard work :)

**Comment by aqueos** [comment]
Thanks. So the restore if going on now but i have lots of critical errors.
I send you the error @davidducos

**Comment by zhangchuangcheng0807** [comment]
> The package is rebuilt now.

HI @davidducos @nerijus 
I just performed a data backup and recovery test on v0.12.7-2, and the test results are as follows:
1. When the names of the source library and the target library are different, data recovery can be performed normally (when myloader specifies the -s and -B options);
2. Specify the --skip-definer option when the myloader command performs data recovery, which can normally replace the "DEFINER=`user`@`host`" keyword in the view creation statement. Fixed issue #815.
Thanks again for your hard work and attention to community issues, and I wish the mydumper open source project better and better.

**Comment by davidducos** [comment]
@zhangchuangcheng0807, so, everything works ok? Can we close it?

@aqueos informed me that he had to use `--serialized-table-creation ` to avoid critical issues, which is expected as some restoration needs to be single threaded, because a known issues that has been discussed [here](https://github.com/mydumper/mydumper/issues/469#issuecomment-968956439).

**Comment by zhangchuangcheng0807** [comment]
> @zhangchuangcheng0807, so, everything works ok? Can we close it?
> 
> @aqueos informed me that he had to use `--serialized-table-creation ` to avoid critical issues, which is expected as some restoration needs to be single threaded, because a known issues that has been discussed [here](https://github.com/mydumper/mydumper/issues/469#issuecomment-968956439).

@davidducos, Yes, everything works fine with the tests now, and all the issues on my side have been resolved. This issue can be closed.
Thanks again.

**Comment by thibaut-pro** [comment]
What are the steps required to make v0.12.7-2 available in Homebrew?
Looks like https://github.com/Homebrew/homebrew-core/blob/master/Formula/mydumper.rb is still on `v0.12.7-1`

**Comment by thibaut-pro** [comment]
Submitted a PR: https://github.com/Homebrew/homebrew-core/pull/111243

**Comment by yourithielen** [comment]
Seeing the same issue here while running command `myloader -u USER -a -d folder_name/` on version `myloader 0.12.7-3, built against MySQL 5.7.35-38`.

I get repeated errors like:
`** (myloader:22365): WARNING **: It was not possible to process file: TABLE-schema.sql.gz (1) because real_db_name isn't found. We might renqueue it, take into account that restores without schema-create files are not supported
`

The server is running Ubuntu 16. Might it have something to do with the default sorting of the files? So maybe OS of shell specific? 

I checked all the tables and the restore seems to be fine.

**Comment by davidducos** [comment]
Hi @yourithielen, 
You can try v0.13.1-1 which is not stable enough for prod, where we changed the behavior of myloader and you should be able to avoid those warnings.
